sdk/identity/identity/src/credentials/azurePipelinesCredential.ts

```diff
@@ -24,40 +24,37 @@ export class AzurePipelinesCredential implements TokenCredential {
 
   /**
    * AzurePipelinesCredential supports Federated Identity on Azure Pipelines through Service Connections.
-   * @param tenantId - tenantId associated with the service connection
-   * @param clientId - clientId associated with the service connection
-   * @param serviceConnectionId - id for the service connection, as found in the querystring's resourceId key
+  
+   * @param systemAccessToken - system access token from the AzurePipelines
    * @param options - The identity client options to use for authentication.
    */
-  constructor(
-    tenantId: string,
-    clientId: string,
-    serviceConnectionId: string,
-    options?: AzurePipelinesCredentialOptions,
-  ) {
+  constructor(systemAccessToken: string, options?: AzurePipelinesCredentialOptions) {
+    const tenantId = options?.tenantId || process.env.AZURESUBSCRIPTION_TENANT_ID;
+    const clientId = options?.clientId || process.env.AZURESUBSCRIPTION_CLIENT_ID;
+    const serviceConnectionId =
+      options?.serviceConnectionId || process.env.AZURESUBSCRIPTION_SERVICE_CONNECTION_ID;
     if (!clientId || !tenantId || !serviceConnectionId) {
```

Comment:
We should also validate that systemAccessToken param is not empty. Yes TypeScript will complain but pure JS users need the runtime validation

Reply:
To be fair, TS doesn't bother doing empty string checks, `""` is still a valid string type, even though it's useless in practice